Nil pointer dereference when shutting down

**Describe the bug**
Runtime panic occurs when visor is shut down manually

**Environment information:**

- Linux
- Platform: `Linux 5.12.4-arch1-2 x86_64`

**Steps to Reproduce**
Steps to reproduce the behavior:

1. Start visor
2. Hit ctrl+C

**Actual behavior**
The following error is shown:
```
[2021-06-02T13:30:03Z] ERROR [transport_setup]: failed to accept error="dmsg error 200 - local entity closed"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x34 pc=0xb78bd9]

goroutine 123 [running]:
github.com/skycoin/dmsg.(*Stream).RawRemoteAddr(...)
	/root/skywire/vendor/github.com/skycoin/dmsg/stream.go:195
github.com/skycoin/skywire/pkg/transport/setup.(*TransportListener).Serve(0xc0003538c0, 0x11d8578, 0xc0002a2680)
	/root/skywire/pkg/transport/setup/visor.go:67 +0x519
created by github.com/skycoin/skywire/pkg/visor.initTransportSetup
	/root/skywire/pkg/visor/init.go:313 +0xdb
```

**Expected behavior**
Shutdown finishes with no runtime panics

**Possible implementation**
Perhaps removed shutdown synchronization is to blame (which was introduced to avoid data races in shutdown). Adding synchronization not based on waitgroups could solve the issue.

--- a/pkg/transport/setup/visor.go
+++ b/pkg/transport/setup/visor.go
@@ -63,6 +63,7 @@ func (ts *TransportListener) Serve(ctx context.Context) {
 		conn, err := lis.AcceptStream()
 		if err != nil {
 			ts.log.WithError(err).Error("failed to accept")
+			break
 		}
 		remotePK := conn.RawRemoteAddr().PK
 		found := false

